content/upcoming/enry.md

```diff
@@ -0,0 +1,206 @@
+
+The number of language detections for each file in samples directory and per each time interval in a logarithmic scale has been measured for both tools, getting the following results:
+
+```
+enry processed files: 1839
+     1us-10us 5.220228%
+     10us-100us 67.645459%
+     100us-1ms 17.455139%
+     1ms-10ms 7.340946%
+     10ms-100ms 2.338227%
+
+linguist processed files: 1839
+     1us-10us 0.000000%
+     10us-100us 4.023926%
+     100us-1ms 50.027189%
+     1ms-10ms 42.849375%
+     10ms-100ms 3.099511%
+```
+
+As you can see, *enry* was able to detect 72% of files in a time between 1us and 100us, while the majority of the files *linguist* processed are shifted to greater time intervals.
+
+Calculating the mean spent time to process a file for both tools, on average *enry* is 211% faster than *linguist*.
+
+Considering that *enry* follows the same algorithms that *linguist* does, it looks like the performance improvement is given by the chosen language to develop them.
+
+However, it should be noted that in a few cases *enry* could turn slower than linguist. This is due to Golang's regexp being slower than Ruby's, which uses [oniguruma](https://github.com/kkos/oniguruma) library, written in C.
+
+## enry CLI
+
+*enry* can be used as a command too
+
+```bash
+$ enry --help
+enry, A simple (and faster) implementation of github/linguist
+usage: enry <path>
+              enry <path> [--json] [--breakdown]
+              enry [--json] [--breakdown]
+```
+
+, and it's programmed to return an output similar to *linguist*'s output.
+
+```bash
+$ enry
+11.11%    Gnuplot
+22.22%    Ruby
+55.56%    Shell
+11.11%    Go
+```
+
+The command has flags to get the results broken down by file,
+
+```bash
+$ enry --breakdown
+11.11%    Gnuplot
+22.22%    Ruby
+55.56%    Shell
+11.11%    Go
+
+Gnuplot
+plot-histogram.gp
+
+Ruby
+linguist-samples.rb
+linguist-total.rb
+
+Shell
+parse.sh
+plot-histogram.sh
+run-benchmark.sh
+run-slow-benchmark.sh
+run.sh
+
+Go
+parser/main.go
+```
+
+and to show it in JSON format,
+
+```bash
+$ enry --json
+{"Gnuplot":["plot-histogram.gp"],"Go":["parser/main.go"],"Ruby":["linguist-samples.rb","linguist-total.rb"],"Shell":["parse.sh","plot-histogram.sh","run-benchmark.sh","run-slow-benchmark.sh","run.sh"]}
+```
+
+The main difference with linguist's command is that *enry* doesn't need a git repository in the directory to analyse it!
+
+## What I really want to know is where "enry" comes from!
+
+In the movie [My Fair Lady](https://en.wikipedia.org/wiki/My_Fair_Lady), [Professor Henry Higgins](http://www.imdb.com/character/ch0011719/?ref_=tt_cl_t2) is one of the main characters. Henry is a linguist and at the very beginning of the movie enjoys guessing the street where people live based on their accent.
```

Comment:
Henry is a linguist and enjoys guessing the street where people live based on their accent in the very beginning of the movie